phpstan-baseline.neon

```diff
@@ -1,32 +1,487 @@
 parameters:
 	ignoreErrors:
+		-
+			message: "#^Method loophp\\\\collection\\\\Collection\\:\\:empty\\(\\) should return loophp\\\\collection\\\\Collection\\<NewTKey, NewT\\> but returns loophp\\\\collection\\\\Collection\\<\\(int&NewTKey\\)\\|\\(NewTKey&string\\), NewT\\>\\.$#"
+			count: 1
+			path: src/Collection.php
+
+		-
+			message: "#^Parameter \\#1 \\$callable of class loophp\\\\collection\\\\Iterator\\\\ClosureIterator constructor expects callable\\(\\.\\.\\.mixed\\)\\: iterable, Closure\\(Generator\\)\\: Generator\\<mixed, mixed, mixed, void\\> given\\.$#"
+			count: 1
+			path: src/Collection.php
+
```

Comment:
@drupol I'm a bit concerned by the high number of new errors this is adding to the baseline. This is likely due to PHPStan's new level 9, which is quite brutal, similar to Psalm's level 1 which disallows mixed (which we are not using - this this project uses the default Psalm level 2).

This PR is currently trying to do 2 things at the same time, and I think we should separate them:
- updating to PHPStan 1.0 -> we should be able to do this and fix any errors as long as there aren't too many

- updating to PHPStan level 9 -> I don't think we can do this yet, it might take quite a bit of work to get the static analysis in place properly for this. For now I think we should limit the level to 8 in the [grumphp](https://github.com/drupol/php-conventions/blob/master/config/psr12/grumphp.yml#L88) config for _this project only_

Reply:
Right, I had this in mind as well at the beginning, then I skipped it.

I will update the PR accordingly in the evening.

Reply:
Job done!